crates/runtime/src/instance/allocator/pooling.rs

```diff
@@ -532,6 +571,47 @@ impl InstancePool {
         Ok(())
     }
 
+    #[cfg(feature = "memfd-allocator")]
+    fn map_instance_memfd_memories(
+        instance_idx: usize,
+        instance: &mut Instance,
+        maybe_memfds: &Option<Arc<ModuleMemFds>>,
+        memories: &MemoryPool,
+        max_wasm_pages: u64,
+    ) -> Result<(), InstantiationError> {
+        let module = instance.module.as_ref();
+
+        debug_assert!(instance.memories.is_empty());
+
+        for (defined_index, plan) in (&module.memory_plans.values().as_slice()
+            [module.num_imported_memories..])
+            .iter()
+            .enumerate()
+        {
+            let memory_index = module.num_imported_memories + defined_index;
+            let mut slot = memories.take_memfd_slot(instance_idx, memory_index)?;
+            let defined_index = DefinedMemoryIndex::from_u32(defined_index as u32);
```

Comment:
Could this use `memory_plans.iter()` and ... I forget the method name but I'm sure we have one to convert from `MemoryIndex` to `DefinedMemoryIndex` ... to avoid using `from_u32`?

Reply:
Ah, yes, I'm using `.values()` here but the default `.iter()` on a `PrimaryMap` gives us an iterator that actually just gives the keys (the strong index type) directly. Much nicer.